pkg/helm/chainlink/chainlink.go

```diff
@@ -151,6 +165,16 @@ func New(index int, props map[string]any) environment.ConnectedChart {
 	return NewVersioned(index, "", props)
 }
 
+// NewDeployment ensures that each chainlink node gets its own helm chart, and should be preferred over New
+// Avoid using replicas when using NewDeployment
```

Comment:
Should we add a check for replicas in this and throw an error since it could create unexpected results?

Reply:
This changes the UX from 

```go
	testEnvironment := environment.New(&environment.Config{
		NamespacePrefix: fmt.Sprintf("upgrade-version-%s", strings.ReplaceAll(strings.ToLower(network.Name), " ", "-")),
		Test:            t,
	}).
		AddHelm(evmConfig).
		AddHelmCharts(chainlink.NewDeployment(3, map[string]any{
			"toml": client.AddNetworksConfig("", network),
			"db": map[string]any{
				"stateful": true,
			},
		}))
	err := testEnvironment.Run()
```

to

```go
	charts, err := chainlink.NewDeployment(3, map[string]any{
		"toml": client.AddNetworksConfig("", network),
		"db": map[string]any{
			"stateful": true,
		},
	})
	require.NoError(t, err)
	testEnvironment := environment.New(&environment.Config{
		NamespacePrefix: fmt.Sprintf("upgrade-version-%s", strings.ReplaceAll(strings.ToLower(network.Name), " ", "-")),
		Test:            t,
	}).
		AddHelm(evmConfig).
		AddHelmCharts(charts)
	err = testEnvironment.Run()
```

so I don't _love_ it, but it can work.